src/saveload/saveload.h

```diff
@@ -401,15 +400,23 @@ enum ChunkType {
 /** Handlers and description of chunk. */
 struct ChunkHandler {
 	uint32 id;                          ///< Unique ID (4 letters).
-	ChunkSaveLoadProc *save_proc;       ///< Save procedure of the chunk.
-	ChunkSaveLoadProc *load_proc;       ///< Load procedure of the chunk.
-	ChunkSaveLoadProc *ptrs_proc;       ///< Manipulate pointers in the chunk.
-	ChunkSaveLoadProc *load_check_proc; ///< Load procedure for game preview.
 	ChunkType type;                     ///< Type of the chunk. @see ChunkType
+
+	ChunkHandler(uint32 id, ChunkType type) : id(id), type(type) {}
+
+	virtual ~ChunkHandler() {}
+
+	virtual void Save() const { NOT_REACHED(); }
+	virtual void Load() const = 0;
+	virtual void FixPointers() const {}
```

Comment:
Save and Load are pretty self-explanatory, though maybe it is worth to explain in some documentation what FixPointers and LoadCheck are doing, and especially when they should be implemented and when implementing them is a waste of effort.
For Save it might be useful to mention that if the chunk is only needed for loading old savegames and it should not be added to the savegame, then it can be left unimplemented but CH_READONLY must be set as chunk type.

Reply:
Based on the old `load_check_proc` comment, maybe `LoadCheck()` should be renamed to `LoadPreview()`.

Reply:
I would leave it as `LoadCheck()` for now. Otherwise you would need to rename everything related to it as well, so something for a future PR.